qos/cosmos/response.go

```diff
@@ -69,7 +52,7 @@ func unmarshalResponse(
 	if err := json.Unmarshal(data, &jsonrpcResponse); err != nil {
 		// The response raw payload could not be unmarshalled as a JSON-RPC response.
 		// Treat it as a REST response and use the generic unmarshaller.
-		return responseUnmarshallerGeneric(logger, jsonrpcResponse, data)
+		return responseUnmarshallerGeneric(logger, jsonrpcResponse, data, isJSONRPC)
```

Comment:
Do we even need `isJSONRPC`?

Feels like this should be `false`, the one below should be `true` and then we don't need to pass in the extra variable?